### PR TITLE
Let the executor create an executor id to its liking.

### PR DESCRIPTION
**Pull request by baldersheim**
Taske full advantage over strands instead of being limited to threads.

@geirst and @havardpe PR

### CODE REVIEW

**Comment by havardpe** [comment]
The following tests FAILED:
	568 - staging_vespalib_adaptive_sequenced_executor_test_app (Failed)

File: staging_vespalib/src/vespa/vespalib/util/adaptive_sequenced_executor.cpp [hunk 1]
```diff
@@ -256,6 +256,11 @@ AdaptiveSequencedExecutor::~AdaptiveSequencedExecutor()
     assert(_worker_stack.empty());
 }
 
+ISequencedTaskExecutor::ExecutorId
+AdaptiveSequencedExecutor::getExecutorId(uint64_t component) const {
```
Comment (havardpe) [inline staging_vespalib/src/vespa/vespalib/util/adaptive_sequenced_executor.cpp:260]: this bypasses a lot of magic implemented in the 'interface'; is that intentional? if yes, move magic to the other subclass. If no, move magic to utility function that can be called from this one.

**Review by havardpe — CHANGES_REQUESTED** [review]
Please fix failing unit test.
Also consider the fact that syncing all strands will be more expensive than before.

File: staging_vespalib/src/vespa/vespalib/util/adaptive_sequenced_executor.cpp (hunk 1 above)
Comment (baldersheim) [reply staging_vespalib/src/vespa/vespalib/util/adaptive_sequenced_executor.cpp:260]: Yes, that is intentional. I planned to move the magic out of the interface in a separate PR. I can put it in this one if you prefer that.

**Comment by baldersheim** [comment]
@havardpe Test has been updated. PTAL

**Review by havardpe — APPROVED** [review]

**Pull request merged**